smart-contracts/test/mainnet/udt.js

```diff
@@ -0,0 +1,371 @@
+      )
+    })
+    describe('the Unlock contract', () => {
+      it('is declared as minter', async () => {
+        assert.equal(await udt.isMinter(unlockAddress), true)
+      })
+      it('can mint', async () => {
+        const recipient = await ethers.Wallet.createRandom()
+
+        await impersonate(unlockAddress)
+        const unlock = await ethers.getSigner(unlockAddress)
+        const tx = await udt.connect(unlock).mint(recipient.address, amount)
+
+        const { events } = await tx.wait()
+        const { args } = events.find((v) => v.event === 'Transfer')
+
+        assert.equal(args.from, ethers.constants.AddressZero)
+        assert.equal(args.to, recipient.address)
+        assert.equal(args.value.eq(amount), true)
+      })
+    })
+  })
+
+  describe('burn', () => {
+    it('function does not exist', async () => {
+      assert.equal(udt.burn, null)
+      assert.equal(Object.keys(udt).includes('burn'), false)
+      assert.equal(Object.keys(udt.functions).includes('burn'), false)
+      assert.equal(Object.keys(udt.functions).includes('burn()'), false)
+    })
+  })
+
+  describe.only('supply', () => {
+    it('is more than 1M', async () => {
+      const totalSupply = await udt.totalSupply()
+      assert.equal(
+        totalSupply.gt(0),
+        true,
+        'starting supply must be different from 0'
+      )
+      // more than 1M
+      assert(totalSupply.gt(ethers.utils.parseEther('1000000')))
+    })
+
+    // totalSupply at block height for ERC20Votes function
+    describe('pastTotalSupply', async () => {
+      it('corresponds to latest totalSupply', async () => {
+        const blockNumber = await ethers.provider.getBlockNumber()
+        await time.advanceBlock()
+        const pastTotalSupply = await udt.getPastTotalSupply(blockNumber)
+        const totalSupply = await udt.totalSupply()
+        assert.isTrue(pastTotalSupply.eq(totalSupply))
+      })
+      it('increases when tokens are minted', async () => {
+        const amount = ethers.utils.hexStripZeros(
+          ethers.utils.parseEther('1000')
+        )
+        const blockNumber = await ethers.provider.getBlockNumber()
+        await time.advanceBlock()
+        const pastTotalSupply = await udt.getPastTotalSupply(blockNumber)
+
+        // mint some tokens
+        const recipient = await ethers.Wallet.createRandom()
+        await impersonate(unlockAddress)
+        const unlock = await ethers.getSigner(unlockAddress)
+        const tx = await udt.connect(unlock).mint(recipient.address, amount)
+        const receipt = await tx.wait()
+        await time.advanceBlock()
+
+        const pastTotalSupplyAfter = await udt.getPastTotalSupply(
+          receipt.blockNumber
+        )
+        assert.isTrue(pastTotalSupplyAfter.eq(pastTotalSupply.add(amount)))
+      })
+    })
+  })
+
+  describe('transfers', () => {
+    it('should support transfer by permit', async () => {
```

Comment:
Let's add a simple test for normal transfers

Reply:
ok added